### PR TITLE
Chore/codebase improvements

### PR DESCRIPTION
**Pull request by phinterseer**
Please review CODEOWNERS.

Let me know if `.nvmrc` is OK.

The teardown-script is now TypeScript ran via ts-node, let me know if there is better way of doing it.

Teardown now also gets rid of Athena workgroups and should hopefully reliably remove your stack in one go.

### CODE REVIEW

**Review by markschnitziusgovuk — DISMISSED** [review]

File: src/tools/teardown-cf-stack.ts [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  DeleteStackCommand,
+  DescribeStackResourcesCommand, StackResource
+} from "@aws-sdk/client-cloudformation";
+import { AthenaClient, DeleteWorkGroupCommand } from "@aws-sdk/client-athena";
+
+interface AWSError { error: any }
+
+const isAWSError = (object: any): object is AWSError => !!object.error
+
+const isStackResource = (object: any): object is StackResource => (object != null && object.ResourceStatus)
```
[the remark this answers is not in the document]
Comment (phinterseer) [reply src/tools/teardown-cf-stack.ts:25]: Ah, I like that. Will do.

File: src/tools/teardown-cf-stack.ts [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  if (isAWSError(result)) {
+    console.log(result.error.toString());
+    process.exit(-1);
+  }
+
+  return (result.StackResources || [])
+    .filter((r) => r.ResourceType === 'AWS::S3::Bucket')
+    .filter((r) => r.ResourceStatus !== 'DELETE_COMPLETE')
+    .map((b) => b.PhysicalResourceId);
+}
```
[the remark this answers is not in the document]
Comment (phinterseer) [reply src/tools/teardown-cf-stack.ts:87]: True, will do that.

File: src/tools/teardown-cf-stack.ts [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+import {
+  DeleteBucketCommand,
+  DeleteObjectCommand,
+  ListObjectVersionsCommand,
+  S3Client
+} from "@aws-sdk/client-s3";
+import {
+  CloudFormationClient,
+  DeleteStackCommand,
+  DescribeStackResourcesCommand, StackResource
```
[the remark this answers is not in the document]
Comment (phinterseer) [reply src/tools/teardown-cf-stack.ts:17]: Oh, I didn't even realise we had husky hooks. Will activate and put it in the README.

**Review by MelAtGDS — DISMISSED** [review]

File: src/tools/teardown-cf-stack.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (phinterseer) [reply src/tools/teardown-cf-stack.ts]: eslint had a lot to complain about. But should all be fixed now.

[the remark this answers is not in the document]
Comment (phinterseer) [reply src/tools/teardown-cf-stack.ts]: Also for some reason eslint was switched off in my IntelliJ...

**Review by MelAtGDS — DISMISSED** [review]

File: README.md [hunk 4]
```diff
@@ -10,6 +10,20 @@ Monitors user identity verification events for billing purposes
 
 If you are new to the Billing & Transaction Monitoring team, please also read [our New Starters guide](https://govukverify.atlassian.net/l/cp/XMirz7JE)
 
+## Do this once
+Install and use the correct node version (using .nvmrc):
+```sh
+nvm install
```
[the remark this answers is not in the document]
Comment (phinterseer) [reply README.md:16]: Done

**Pull request merged**